substrate/frame/ranked-collective/src/lib.rs

```diff
@@ -650,6 +658,37 @@ pub mod pallet {
 				pays_fee: Pays::No,
 			})
 		}
+
+		/// Exchanges a member with a new account and the same existing rank.
+		///
+		/// - `origin`: Must be the `ExchangeOrigin`.
+		/// - `who`: Account of existing member of rank greater than zero to be exchanged.
+		/// - `new_who`: New Account of existing member of rank greater than zero to exchanged to.
+		#[pallet::call_index(6)]
+		#[pallet::weight(T::WeightInfo::exchange_member())]
+		pub fn exchange_member(
+			origin: OriginFor<T>,
+			who: AccountIdLookupOf<T>,
+			new_who: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			T::ExchangeOrigin::ensure_origin(origin)?;
+			let who = T::Lookup::lookup(who)?;
+			let new_who = T::Lookup::lookup(new_who)?;
+
+			ensure!(who != new_who, Error::<T, I>::SameMember);
+
+			let MemberRecord { rank, .. } = Self::ensure_member(&who)?;
+
+			for r in 0..=rank {
```

Comment:
If you make an equivalent of `do_remove_member_to_rank` that does these two operations, it would be a good counterpart to `do_add_member_to_rank`, it would make the code a bit nicer, and you can use it in other places as well.